src/test/rgw/test_d4n_filter.cc

```diff
@@ -0,0 +1,2810 @@
+      destObjEnabled->gen_rand_obj_instance_name();
+      instance = destObjEnabled->get_instance();
+
+      int ret = objEnabled->copy_object(acl_owner,
+	       std::get<rgw_user>(owner),
+	       &info,
+	       zone,
+	       destObjEnabled.get(),
+	       testBucket.get(),
+	       testBucket.get(),
+	       placement,
+	       &mtime,
+	       &mtime,
+	       nullptr,
+	       nullptr,
+	       false,
+	       nullptr,
+	       nullptr,
+	       rgw::sal::ATTRSMOD_NONE,
+	       false,
+	       attrs, 
+	       RGWObjCategory::Main,
+	       0,
+	       boost::none,
+	       nullptr,
+	       &tag, 
+	       &tag,
+	       nullptr,
+	       nullptr,
+	       env->dpp,
+	       optional_yield({yield}));
+      EXPECT_EQ(ret, 0);
+
+      boost::system::error_code ec;
+      request req;
+      req.push("EXISTS", TEST_BUCKET + testName + "__:" + instance + "_" + destNameEnabled + "_0_0");
+      req.push("HGETALL", TEST_BUCKET + testName + "__:" + instance + "_" + destNameEnabled + "_0_0");
+
+      response< int, std::map<std::string, std::string> > resp;
+
+      conn->async_exec(req, resp, yield[ec]);
+
+      ASSERT_EQ((bool)ec, false);
+      EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+      EXPECT_EQ(std::get<1>(resp).value().size(), 10);
+
+      std::error_code err;
+      EXPECT_EQ(fs::exists(CACHE_DIR + "/" + TEST_BUCKET + testName + "/" + destNameEnabled + "/" + instance, err), true);  
+    }
+
+    put_version_suspended_object(testName, yield);
+
+    {
+      std::string destNameSuspended = "dest_object_suspended";
+      std::unique_ptr<rgw::sal::Object> destObjSuspended = testBucket->get_object(rgw_obj_key(destNameSuspended));
+      EXPECT_NE(destObjSuspended.get(), nullptr);
+
+      int ret = objSuspended->copy_object(acl_owner,
+	       std::get<rgw_user>(owner),
+	       &info,
+	       zone,
+	       destObjSuspended.get(),
+	       testBucket.get(),
+	       testBucket.get(),
+	       placement,
+	       &mtime,
+	       &mtime,
+	       nullptr,
+	       nullptr,
+	       false,
+	       nullptr,
+	       nullptr,
+	       rgw::sal::ATTRSMOD_NONE,
+	       false,
+	       attrs, 
+	       RGWObjCategory::Main,
+	       0,
+	       boost::none,
+	       nullptr,
+	       &tag, 
+	       &tag,
+	       nullptr,
+	       nullptr,
+	       env->dpp,
+	       optional_yield({yield}));
+      EXPECT_EQ(ret, 0);
+
+      boost::system::error_code ec;
+      request req;
+      req.push("EXISTS", TEST_BUCKET + testName + "__:null_" + destNameSuspended + "_0_0");
+      req.push("HGETALL", TEST_BUCKET + testName + "__:null_" + destNameSuspended + "_0_0");
+      req.push("FLUSHALL");
+
+      response< int, std::map<std::string, std::string>,
+	       boost::redis::ignore_t > resp;
+
+      conn->async_exec(req, resp, yield[ec]);
+
+      ASSERT_EQ((bool)ec, false);
+      EXPECT_EQ((int)std::get<0>(resp).value(), 1);
+      EXPECT_EQ(std::get<1>(resp).value().size(), 10);
+
+      std::error_code err;
```

Comment:
similar comment as above - wherever you test copyObject with different xattrs flag, you can add checks to see if the result is what you expect them to be.